File: backend/src/main/java/com/mftb/admin/entity/EamBorrow.java
package com.mftb.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 资产借用登记实体
 */
@Data
@TableName("biz_eam_borrow")
public class EamBorrow {

    @TableId
    private Long id;

    /** 借用编号（系统生成，唯一） */
    private String borrowNo;

    /** 资产 ID */
    private Long assetId;

    /** 借用人 ID */
    private Long holderId;

    /** 借用人姓名快照 */
    private String holderName;

    /** 借用部门 */
    private String department;

    /** 操作人 ID */
    private Long operatorId;

    /** 操作人姓名快照 */
    private String operatorName;

    /** 状态：active/overdue/returned/cancelled */
    private String status;

    /** 借出日期 */
    private LocalDate startDate;

    /** 到期日期 */
    private LocalDate dueDate;

    /** 实际归还日期 */
    private LocalDate returnDate;

    /** 借用用途 */
    private String purpose;

    /** 续借次数 */
    private Integer renewCount;

    /** 关联归还记录 ID */
    private Long returnId;

    /** 创建人 */
    private String createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /** 最后更新人 */
    private String updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: src/api/eamBorrow.ts
/**
 * 借用管理 API
 *
 * 后端接口: /api/eam/borrows/*
 */
import request, { isBackendUnavailable } from './request'

/* ==================== 类型定义 ==================== */

export interface BorrowRow {
  id: number
  borrowNo: string
  assetId: number
  assetNo: string
  assetName: string
  holderId: number
  holderName: string
  department: string
  operatorName: string
  status: 'active' | 'overdue' | 'returned' | 'cancelled'
  startDate: string
  dueDate: string
  returnDate?: string
  purpose?: string
  renewCount: number
  returnId?: number
  createdAt: string
  updatedAt: string
  overdueDays?: number
}

export interface BorrowQuery {
  page: number
  size: number
  keyword?: string
  status?: string
  department?: string
  startDate?: string
  endDate?: string
}

export interface BorrowPage<T> {
  records: T[]
  total: number
}

export interface BorrowRegisterDTO {
  assetId: number
  holderId: number
  department: string
  startDate: string
  dueDate: string
  purpose?: string
}

export interface BorrowRenewDTO {
  newDueDate: string
}

/* ==================== API 方法 ==================== */

/** 分页查询 */
export async function fetchBorrowList(query: BorrowQuery): Promise<BorrowPage<BorrowRow>> {
  try {
    const params = new URLSearchParams()
    params.set('page', String(query.page))
    params.set('size', String(query.size))
    if (query.keyword) params.set('keyword', query.keyword)
    if (query.status) params.set('status', query.status)
    if (query.department) params.set('department', query.department)
    if (query.startDate) params.set('startDate', query.startDate)
    if (query.endDate) params.set('endDate', query.endDate)
    return await request.get<unknown, BorrowPage<BorrowRow>>(`/eam/borrows?${params}`)
  } catch (err) {
    if (isBackendUnavailable(err)) {
      return { records: [], total: 0 }
    }
    throw err
  }
}

/** 详情 */
export async function fetchBorrowDetail(id: number): Promise<BorrowRow> {
  return request.get<unknown, BorrowRow>(`/eam/borrows/${id}`)
}

/** 登记借用 */
export async function registerBorrow(dto: BorrowRegisterDTO): Promise<number> {
  return request.post<unknown, number>('/eam/borrows', dto)
}

/** 续借 */
export async function renewBorrow(id: number, dto: BorrowRenewDTO): Promise<void> {
  await request.post<unknown, void>(`/eam/borrows/${id}/renew`, dto)
}

/** 取消借用 */
export async function cancelBorrow(id: number, reason: string): Promise<void> {
  await request.post<unknown, void>(`/eam/borrows/${id}/cancel`, { reason })
}
